src/conn/conn_sweep.c

```diff
@@ -129,32 +111,110 @@ __sweep(WT_SESSION_IMPL *session)
 		    !__wt_txn_visible_all(session, btree->rec_max_txn))
 			goto unlock;
 
-		/* If the handle is open, try to close it. */
-		if (F_ISSET(dhandle, WT_DHANDLE_OPEN)) {
+		if (F_ISSET(dhandle, WT_DHANDLE_OPEN))
 			WT_WITH_DHANDLE(session, dhandle, ret =
-			    __wt_conn_btree_sync_and_close(session, 0, 0));
-
-			/* We closed the btree handle, bump the statistic. */
-			if (ret == 0)
-				WT_STAT_FAST_CONN_INCR(
-				    session, dh_conn_handles);
-		}
-
-		if (dhandle->session_inuse == 0 && dhandle->session_ref == 0)
-			++closed_handles;
+			    __wt_conn_btree_sync_and_close(session, 0, 1));
```

Comment:
Is it possible to have a handle w/o WT_DHANDLE_OPEN set here? If so, can't we just mark it WT_DHANDLE_DEAD and flush it?